client_test.go

```diff
@@ -3182,3 +3192,61 @@ func Test_AddMissingPort(t *testing.T) {
 		})
 	}
 }
+
+type TransportEx struct {
+	base    RoundTripper
+	reqLog  *string
+	respLog *string
+}
+
+func (t TransportEx) RoundTrip(hc *HostClient, req *Request, resp *Response) (bool, error) {
+	req.Header.Set("trace-id", "123")
+	*t.reqLog = req.String()
+	retry, err := t.base.RoundTrip(hc, req, resp)
+	*t.respLog = resp.String()
+	return retry, err
+}
+
+func TestClientTransportEx(t *testing.T) {
+	var reqLog string
+	var respLog string
```

Comment:
Why not create `TransportEx` here, change it to `func (t *TransportEx) RoundTrip` and set `hc.Transport = &tp` so you don't have to use string pointers?

Reply:
I have rewrite the testacase. Please check again. Thanks!
Using interfaces is more extensible than using methods, and it is easier to add methods to RoundTripper in the future.